scripts/migration/migrate-istio

```diff
@@ -0,0 +1,1360 @@
+  ROLES="$(\
+    kubectl get clusterrolebinding \
+    --all-namespaces \
+    -o jsonpath='{range .items[?(@.subjects[0].name=="'"${GCLOUD_USER}"'")]}[{.roleRef.name}]{end}'\
+    2>/dev/null)"
+  if echo "${ROLES}" | grep -q cluster-admin; then return; fi
+
+  ROLES="$(gcloud projects \
+    get-iam-policy "${PROJECT_ID}" \
+    --flatten='bindings[].members' \
+    --filter="bindings.members:${IAM_USER}" \
+    --format='value(bindings.role)' 2>/dev/null)"
+  if echo "${ROLES}" | grep -q roles/container.admin; then return; fi
+  if echo "${ROLES}" | grep -q roles/owner; then return; fi
+
+  info "No admin role found for current user."
+  false
+}
+
+bind_user_to_cluster_admin(){
+  info "Querying for core/account..."
+  local GCLOUD_USER; GCLOUD_USER="$(gcloud config get-value core/account)"
+  info "Binding ${GCLOUD_USER} to cluster admin role..."
+  local PREFIX; PREFIX="$(echo "${GCLOUD_USER}" | cut -f 1 -d @)"
+  local YAML; YAML="$(retry 2 run kubectl create \
+    clusterrolebinding "${PREFIX}-cluster-admin-binding" \
+    --clusterrole=cluster-admin \
+    --user="${GCLOUD_USER}" \
+    --dry-run -o yaml)"
+  retry 3 run kubectl apply -f - <<EOF
+${YAML}
+EOF
+  info "Successfully bound to cluster admin role."
+}
+
+exit_if_not_cluster_admin() {
+  if ! is_user_cluster_admin; then
+    { read -r -d '' MSG; fatal "${MSG}"; } <<EOF || true
+Current user must have the cluster-admin role on ${CLUSTER_NAME}.
+Please add the cluster role binding and retry, or run the script with the
+'--enable-dependencies' flag to allow the script to enable it on your behalf.
+EOF
+  fi
+}
+
+check_dependencies() {
+  exit_if_apis_not_enabled
+  exit_if_cluster_unregistered
+  exit_if_service_mesh_feature_not_enabled
+  exit_if_no_workload_identity
+  exit_if_stackdriver_not_enabled
+  exit_if_cluster_unlabeled
+  exit_if_not_cluster_admin
+}
+
+exit_if_apis_not_enabled() {
+  local ENABLED; ENABLED="$(get_enabled_apis)";
+  local REQUIRED; REQUIRED="$(required_apis)";
+  local NOTFOUND; NOTFOUND="";
+
+  info "Checking required APIs..."
+  NOTFOUND="$(find_missing_strings "${REQUIRED}" "${ENABLED}")"
+
+  if [[ -n "${NOTFOUND}" ]]; then
+    for api in $(echo "${NOTFOUND}" | tr ' ' '\n'); do
+      warn "API not enabled - ${api}"
+    done
+    { read -r -d '' MSG; fatal "${MSG}"; } <<EOF || true
+One or more APIs are not enabled. Please enable them and retry, or run the
+script with the '--enable-dependencies' flag to allow the script to enable them on
+your behalf.
+EOF
+  fi
+}
+
+validate_cluster() {
+  validate_cluster_name
+  validate_node_pool
+  validate_istio_version
+}
+
+list_valid_pools() {
+  gcloud container node-pools list \
+    --project="${PROJECT_ID}" \
+    --region "${CLUSTER_LOCATION}" \
+    --cluster "${CLUSTER_NAME}" \
+    --filter "$(valid_pool_query "${1}")"\
+    --format=json
+}
+
+valid_pool_query() {
+  cat <<EOF | tr '\n' ' '
+    config.machineType.split(sep="-").slice(-1:) >= $1
+EOF
+}
+
+validate_node_pool() {
```

Comment:
nit: leaving this as a remainder to add this to install_asm, if not already added